### ISSUE TITLE
Code does not converge, tf 1.0.1 and tf 0.12.1

### ISSUE THREAD

**Issue by Amir-Ramezani** [apoorv2904/DDPG-Torcs-Tensorflow#2]
Hello

I ran your code several times using tf 1.0.1 and tf 0.12.1. Unfortunately it does not converge to an optimal policy. I got the following results at different times where it shows the actions are saturated. I appreciate if you help me to find where is the problem.

('Episode', 40, 'Step', 834.0, 'Epsilon', 0.8291466666655474, 'Action', array([ -1.00000000e+00,   2.88813135e-19,   1.00000000e+00], dtype=float32), 'Reward', 0.028649471836707862)
('Episode', 40, 'Step', 849.0, 'Epsilon', 0.8291466666655474, 'Action', array([ -1.00000000e+00,   3.35174825e-19,   1.00000000e+00], dtype=float32), 'Reward', -0.11828832494059233)
('Episode', 40, 'Step', 864.0, 'Epsilon', 0.8291466666655474, 'Action', array([ -1.00000000e+00,   3.60342596e-19,   1.00000000e+00], dtype=float32), 'Reward', 0.11092526499916951)
('Episode', 40, 'Step', 879.0, 'Epsilon', 0.8291466666655474, 'Action', array([ -1.00000000e+00,   3.98099798e-19,   1.00000000e+00], dtype=float32), 'Reward', 0.048640735427699894)
('Episode', 40, 'Step', 894.0, 'Epsilon', 0.8291466666655474, 'Action', array([ -1.00000000e+00,   4.38389491e-19,   1.00000000e+00], dtype=float32), 'Reward', -0.060516384307772106)
('Episode', 40, 'Step', 909.0, 'Epsilon', 0.8291466666655474, 'Action', array([ -1.00000000e+00,   4.74080784e-19,   1.00000000e+00], dtype=float32), 'Reward', -0.046591250868712193)
('Episode', 40, 'Step', 924.0, 'Epsilon', 0.8291466666655474, 'Action', array([ -1.00000000e+00,   4.85053024e-19,   1.00000000e+00], dtype=float32), 'Reward', 0.058505236735221712)
('Episode', 40, 'Step', 939.0, 'Epsilon', 0.8291466666655474, 'Action', array([ -1.00000000e+00,   5.31340345e-19,   1.00000000e+00], dtype=float32), 'Reward', 0.072708157203574417)
('Episode', 40, 'Step', 954.0, 'Epsilon', 0.8291466666655474, 'Action', array([ -1.00000000e+00,   5.56986512e-19,   1.00000000e+00], dtype=float32), 'Reward', -0.035003065107430828)
('Episode', 40, 'Step', 969.0, 'Epsilon', 0.8291466666655474, 'Action', array([ -1.00000000e+00,   6.58464667e-19,   1.00000000e+00], dtype=float32), 'Reward', 0.026907197476024769)
('Episode', 40, 'Step', 984.0, 'Epsilon', 0.8291466666655474, 'Action', array([ -1.00000000e+00,   6.80764940e-19,   1.00000000e+00], dtype=float32), 'Reward', 0.028280042767128491)
('Episode', 40, 'Step', 999.0, 'Epsilon', 0.8291466666655474, 'Action', array([ -1.00000000e+00,   6.34610433e-19,   1.00000000e+00], dtype=float32), 'Reward', 0.00095780650206023622)
('Episode', 40, 'Step', 1014.0, 'Epsilon', 0.8291466666655474, 'Action', array([ -1.00000000e+00,   8.00504762e-19,   1.00000000e+00], dtype=float32), 'Reward', -0.16836338915552615)
('Episode', 40, 'Step', 1029.0, 'Epsilon', 0.8291466666655474, 'Action', array([ -1.00000000e+00,   8.88721816e-19,   1.00000000e+00], dtype=float32), 'Reward', 0.094671512364097654)

---
('Episode', 30, 'Step', 1512.0, 'Epsilon', 0.8592399999990779, 'Action', array([ -1.00000000e+00,   4.27206485e-18,   1.00000000e+00], dtype=float32), 'Reward', 0.040875252481510436)
('Episode', 30, 'Step', 1527.0, 'Epsilon', 0.8592399999990779, 'Action', array([ -1.00000000e+00,   4.51666050e-18,   1.00000000e+00], dtype=float32), 'Reward', 0.012764681301115402)
('Episode', 30, 'Step', 1542.0, 'Epsilon', 0.8592399999990779, 'Action', array([ -1.00000000e+00,   4.56995327e-18,   1.00000000e+00], dtype=float32), 'Reward', -0.099940184609601038)
('Episode', 30, 'Step', 1557.0, 'Epsilon', 0.8592399999990779, 'Action', array([ -1.00000000e+00,   4.96821302e-18,   1.00000000e+00], dtype=float32), 'Reward', 0.0078738738303673235)
('Episode', 30, 'Step', 1572.0, 'Epsilon', 0.8592399999990779, 'Action', array([ -1.00000000e+00,   5.20275057e-18,   1.00000000e+00], dtype=float32), 'Reward', 0.053144923653363478)
('Episode', 30, 'Step', 1587.0, 'Epsilon', 0.8592399999990779, 'Action', array([ -1.00000000e+00,   5.03708408e-18,   1.00000000e+00], dtype=float32), 'Reward', 0.062004167579573805)
('Episode', 30, 'Step', 1602.0, 'Epsilon', 0.8592399999990779, 'Action', array([ -1.00000000e+00,   5.15913251e-18,   1.00000000e+00], dtype=float32), 'Reward', -0.058061630394771399)
('Episode', 30, 'Step', 1617.0, 'Epsilon', 0.8592399999990779, 'Action', array([ -1.00000000e+00,   5.32952469e-18,   1.00000000e+00], dtype=float32), 'Reward', 0.010017109676356259)
('Episode', 30, 'Step', 1632.0, 'Epsilon', 0.8592399999990779, 'Action', array([ -1.00000000e+00,   5.85320322e-18,   1.00000000e+00], dtype=float32), 'Reward', 0.0026021518031508432)

Thanks,

**Comment by fangthu** [comment]
@AmirCognitive hi. I am using DDPG to solve my problem recently. 
I also met with the issue that the actor network is easy is reach the upper bound or lower bound and keep constant. Did you solve this problem yet ?

**Comment by m5823779** [comment]
Hello, I meet same problem, when my action dimension equal to one, the result is good, but when my action dimension is two (the activation function is tanh and sigmoid), the output of actor will saturate. By the way, I use batch normalization only in my actor network.  And my code reference from: https://github.com/floodsung/DDPG
Have you already solve this problem?
